src/latentis/measure/cka.py

```diff
@@ -0,0 +1,190 @@
+        """
+        Compute HSIC for linear kernels.
+
+        This method is used in the computation of linear CKA.
+
+        Args:
+            X, Y: Input matrices.
+
+        Returns:
+            The computed HSIC value.
+        """
+
+        # inter-sample similarity matrices for both spaces ~(N, N)
+        L_X = X @ X.T
+        L_Y = Y @ Y.T
+
+        return torch.sum(self.centering(L_X) * self.centering(L_Y))
+    
+    def kernel_HSIC(self, X: torch.Tensor, Y:torch.Tensor, sigma):
+        """
+        Compute HSIC (Hilbert-Schmidt Independence Criterion) for RBF kernels.
+
+        This is used in the computation of kernel CKA.
+
+        Args:
+            X, Y: Input matrices.
+            sigma: The RBF kernel width.
+
+        Returns:
+            The computed HSIC value.
+        """
+        return torch.sum(self.centering(self.rbf(X, sigma)) * self.centering(self.rbf(Y, sigma)))
+
+    def centering(self, K: torch.Tensor) -> torch.Tensor:
+        """
+        Center the kernel matrix K using the centering matrix H = I_n - (1/n) 1 * 1^T. (Eq. 3 in the paper)
+
+        This method is used in the calculation of HSIC.
+
+        Args:
+            K: The kernel matrix to be centered.
+
+        Returns:
+            The centered kernel matrix.
+        """
+        n = K.shape[0]
+        unit = torch.ones([n, n]).type_as(K) 
+        identity_mat = torch.eye(n).type_as(K)
+        H = identity_mat - unit / n
+
+        return H @ K @ H
+
+    def rbf(self, X: torch.Tensor, sigma=None):
+        """
+        Compute the RBF (Radial Basis Function) kernel for a matrix X.
+
+        If sigma is not provided, it is computed based on the median distance.
+
+        Args:
+            X: The input matrix.
+            sigma: Optional parameter to specify the RBF kernel width.
+
+        Returns:
+            The RBF kernel matrix.
+        """
+
+        GX = X @ X.T
+        KX = torch.diag(GX) - GX + (torch.diag(GX) - GX).T
+        
+        if sigma is None:
+            mdist = torch.median(KX[KX != 0])
+            sigma = math.sqrt(mdist)
+        
+        KX *= -0.5 / (sigma * sigma)
+        KX = torch.exp(KX)
+
+        return KX
+
+def CKAFn(space1: torch.Tensor, space2: torch.Tensor, mode: CKAMode, device: torch.device = None, sigma=None):
+    """
+    Compute the Centered Kernel Alignment (CKA) between two spaces.
+
+    Args:
+        space1: First embedding matrix or LatentSpace object.
+        space2: Second embedding matrix or LatentSpace object.
+        mode: The mode of CKA (CKAMode.LINEAR or CKAMode.RBF).
+        device: The torch device (e.g., CPU or GPU) to perform calculations on. Defaults to None.
+        sigma: Optional parameter for RBF kernel. Only used when mode is CKAMode.RBF.
+
+    Returns:
+        The computed CKA value.
+    """
+    cka = CKA(mode, device)
+
+    return cka._forward(space1, space2, sigma)
```

Comment:
I think it should be the other way around -- 

- The functional API should implement the functionality (especially when there is not a state to be mantained)
- The object API should mantain the state and call the functional API 

If I am not mistaken that is how `torch.nn.functional` is organized

Reply:
So basically we could have:


`from latentis.measue import CKA`

and 

`from latentis.measure.functions import linear_cka, rbf_cka`

Whad do you think? @Flegyas @crisostomi

Reply:
agreed, will do this way